util/settings/resources_filter.go

```diff
@@ -65,12 +65,34 @@ func (rf *ResourcesFilter) isExcludedResource(apiGroup, kind, cluster string) bo
 //
 func (rf *ResourcesFilter) IsExcludedResource(apiGroup, kind, cluster string) bool {
 	// if excluded, do not allow
-	if rf.isExcludedResource(apiGroup, kind, cluster) {
+	if rf.isExcludedResource(apiGroup, kind, cluster, map[string]string{}) {
 		return true
 	}
 
 	// if included, do allow
-	if rf.isIncludedResource(apiGroup, kind, cluster) {
+	if rf.isIncludedResource(apiGroup, kind, cluster, map[string]string{}) {
+		return false
+	}
+
+	// if inclusion rules defined for cluster, default is not allow
+	for _, includedResource := range rf.ResourceInclusions {
+		if includedResource.MatchCluster(cluster) {
+			return true
+		}
+	}
+
+	// if no inclusion rules defined for cluster, default is allow
+	return false
+}
+
+func (rf *ResourcesFilter) IsExcludedResourceWithAnnotation(apiGroup, kind, cluster string, annotation map[string]string) bool {
```

Comment:
`IsExcludedResource` is used in `gitops-engine`: https://github.com/argoproj/gitops-engine/blob/d25b8fd69f0dc83b0eead1154c5d33add1c61de7/pkg/utils/kube/resource_filter.go#L4
So I can't change arguments of `IsExcludedResource`, I add another method for annotations.